tools/openmm/pdbfixer.xml

```diff
@@ -0,0 +1,189 @@
+            </param>
+            <param name="ionic_strength" type="float" min="0" max="10" value="0" label="Ionic strength" help="Molar concentration of ions to add to the water box"/>
+        </section>
+    </inputs>
+    <outputs>
+        <data name="output" format="pdb"/>
+    </outputs>
+    <tests>
+        <!-- pdbfixer test-data/broken.pdb -add-atoms=all -keep-heterogens=all -replace-nonstandard -->
+        <test>
+            <param name="pdb_input" value="broken.pdb"/>
+            <param name="add_atoms" value="all"/>
+            <param name="keep_heterogens" value="all"/>
+            <param name="replace_nonstandard" value="true"/>
+            <param name="add_residues" value="false"/>
+            <param name="ph" value="7"/>
+            <output name="output">
+                <assert_contents>
+                    <has_n_lines n="162"/>
+                    <!-- check LYS8 is fixed  -->
+                    <has_text text="CD  LYS A   8"/>
+                    <has_text text="CE  LYS A   8"/>
+                    <!-- check hydration -->
+                    <has_text_matching expression="H  \n" n="81"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!-- pdbfixer test-data/broken.pdb -add-atoms=heavy -keep-heterogens=all -replace-nonstandard -add-residues -->
+        <test>
+            <param name="pdb_input" value="broken.pdb"/>
+            <param name="add_atoms" value="heavy"/>
+            <param name="keep_heterogens" value="all"/>
+            <param name="replace_nonstandard" value="true"/>
+            <param name="add_residues" value="false"/>
+            <output name="output">
+                <assert_contents>
+                    <has_n_lines n="81"/>
+                    <!-- check LYS8 is fixed  -->
+                    <has_text text="CD  LYS A   8"/>
+                    <has_text text="CE  LYS A   8"/>
+                    <!-- check no hydration -->
+                    <has_text_matching expression="H  \n" negate="true"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test>
+            <param name="pdb_input" value="broken.pdb"/>
+            <param name="add_atoms" value="all"/>
+            <param name="keep_heterogens" value="all"/>
+            <param name="replace_nonstandard" value="true"/>
+            <param name="add_residues" value="false"/>
+            <param name="ph" value="14"/>
+            <param name="water_box" value="true"/>
+            <param name="x" value="5"/>
+            <param name="y" value="5"/>
+            <param name="z" value="5"/>
+            <param name="positive_ion" value="Li+"/>
+            <param name="negative_ion" value="I-"/>
+            <param name="ionic_strength" value="2"/>
+            <output name="output">
+                <assert_contents>
+                    <!-- seems insertion of water is random and not fully reproducible  -->
```

Comment:
probably a random seed everytime you run it. i think gmx solvate does the same. total number added should be the same maybe, even if coordinates change.

Reply:
Yep, hence the `has_n_lines` assert works